contracts/blade/ChildERC1155Predicate.sol

```diff
@@ -239,12 +262,26 @@ contract ChildERC1155Predicate is IChildERC1155Predicate, Predicate, Initializab
         emit ERC1155WithdrawBatch(rootToken, address(childToken), msg.sender, receivers, tokenIds, amounts);
     }
 
-    function _deposit(bytes calldata data) private {
+    function _withdrawRollback(bytes calldata data) private{
+        (address depositToken, address withdrawer, , uint256 tokenId, uint256 amount) = abi.decode(
+            data,
+            (address, address, address, uint256, uint256)
+        );
+
+        _depositInternal(depositToken, withdrawer, withdrawer, tokenId, amount);
```

Comment:
This is not correct. Where is the `receiver` address. You are decoding it, but not using it. It should be:

```solidity
(address depositToken, address sender, address receiver, uint256 tokenId, uint256 amount) = abi.decode(
       data,
       (address, address, address, uint256, uint256)
);

_depositInternal(depositToken, receiver, sender, tokenId, amount);
```